### PR TITLE
PriorityQueue

### PR DESCRIPTION
**Pull request by AbhirajBose**
#### Description of Change

This is a pull request regarding the Implementation of Priority Queue using Linked List.In this file I have made a code regarding the Priority Queue.Priority Queue is a type of Data Structure which uses the concept of LinkedList to arrange the data in order of priority as instructed by the user.
Contributors guide: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md
-->

#### Checklist

- [ ]Implementation of Priority Queue using Linked List 
- [x] priorityqueue.cpp(https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ]Test Cases
1.Insert
2.Remove
3.Display
4.Exit
Enter your choice
1
Enter the value to be inserted
33
Enter the priority
2
Enter your choice
1 
Enter the value to be inserted
56
Enter the priority
1
Enter your choice
3
Value: 56
Priority: 1
Value: 33
Priority: 2
Enter your choice
4
- [x] Added documentation and the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Searched previous suggestions before making a new one.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Priority queue is a type of data structure which sorts the Linked List in order as provided by the user.Here is the code for the following

### CODE REVIEW

**Comment by AbhirajBose** [comment]
@Panquesito7 ans @realstealthninja Please check my PR and if possible merge it.

**Comment by NitishPal2013** [comment]
@AbhirajBose your PR message has numbers instead of checks in the checkbox. kindly read the markdown down syntax for the adding the checkbox.

Great work by the way but need some more work on this. These are some suggestions:

1) You have not added any test cases. Please add the test cases.
2) Do add the docs about your code. It will help other fellow developers to understand it.
3) Do read the contribution guidelines for the correct format of code and the PR.

**Comment by AbhirajBose** [comment]
@NitishPal2013 Please check it out once now

**Comment by NitishPal2013** [comment]
@AbhirajBose , it still does not follow the [contribution guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/6376bf46aff52a1944d4efcd5ae884b39119cd34/CONTRIBUTING.md). ping me in the discord community general server if you have any problem in understanding it. My discord username nitish_pal .

**Review by MogomotsiFM — CHANGES_REQUESTED** [review]
- Add a function to retrieve one item from the priority queue for possible processing: that is, int PQ::front()
- Almost all your comments are not necessary because your code is easy to follow,
- I would rename the function PQ::remove to PQ::clear. That is what the std library names functions that erase all the contents of a container.
- If you are familiar with std::shared_ptr and std::make_shared use them instead of raw pointers. Otherwise, keep the raw pointers.

**Comment by AbhirajBose** [comment]
@MogomotsiFM Please check my PR now

File: data_structures/priority_queue.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ *
+ */
+#include <iostream>
+using namespace std;
+/*A structure for Node*/
+struct Node {      // defining node of a LinkedList
+    int data;      // defining data part of a node
+    int priority;  // defining priority part of a node
+    Node* next;    // Node pointer points to next
+};
```
Comment (NitishPal2013) [inline data_structures/priority_queue.cpp:21]: Define your structure above from it. 
One more thing whenever we write code, we try that our code must be self explanatory by choosing appropriate variable name and flow of code, which is fine with your code. So, the comments you have added are not necessary here. This is same thing which @MogomotsiFM was highlighting in the suggestions.

File: data_structures/priority_queue.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    Node* front;       // Node pointer points to next
+ public:               // can be accessed globally
+    PQ() {             // constructor
+        front = NULL;  // initializing front as NULL
+    }
+    void insert(int, int);  // calling insert function
+    void clear();           // calling remove function
+    bool isEmpty();         // calling isEmpty function
+    void display();         // calling display function
+};
```
Comment (NitishPal2013) [inline data_structures/priority_queue.cpp:34]: Describe the same above from it.

File: data_structures/priority_queue.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    }
+}
+void PQ::clear()  // using scope resolution operator to use the class variables
+{
+    while (!isEmpty())  // checking if the list is empty or not
+    {
+        Node* temp =
+            front;  // Node pointer pointing to temp is assigned as front
+        front = front->next;  // changing the pointer
+        delete temp;          // deleting the temporary variable
```
Comment (NitishPal2013) [inline data_structures/priority_queue.cpp:67]: All the comments are not unnecessary. You should remove them.

**Review by NitishPal2013 — COMMENTED** [review]
@AbhirajBose do read the suggestions from @MogomotsiFM again. You are on the right way but still need some work.
Remove the unnecessary comments and kindly follow the Doxygen format to document your code.

**Comment by AbhirajBose** [comment]
@NitishPal2013 Please check it now

**Review by MogomotsiFM — APPROVED** [review]
Looks good to me. 

I hope you think that the code is cleaner and not cluttered without comments.

**Comment by AbhirajBose** [comment]
@MogomotsiFM Yes,it is.If possible,please merge my PR.

**Pull request closed without merging**